sdk/core/Azure.Core.TestFramework/src/TestProxy.cs

```diff
@@ -0,0 +1,218 @@
+                s_dotNetExe,
+                proxyPath)
+            {
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                EnvironmentVariables =
+                {
+                    ["ASPNETCORE_URLS"] = $"http://{IpAddress}:0;https://{IpAddress}:0",
+                    ["Logging__LogLevel__Microsoft"] = "Information"
+                }
+            };
+
+            if (!TestEnvironment.GlobalIsRunningInCI)
+            {
+                ImportDevCertIfNeeded();
+            }
+
+            _testProxyProcess = Process.Start(testProxyProcessInfo);
+            ProcessTracker.Add(_testProxyProcess);
+            _ = Task.Run(
+                () =>
+                {
+                    while (!_testProxyProcess.HasExited && !_testProxyProcess.StandardError.EndOfStream)
+                    {
+                        var error = _testProxyProcess.StandardError.ReadLine();
+                        throw new TestProxyException($"An error occurred in the test proxy: {error}");
+                    }
+                });
+
+            int lines = 0;
+            while ((_proxyPortHttp == null || _proxyPortHttps == null) && lines++ < 20)
+            {
+                string outputLine = _testProxyProcess.StandardOutput.ReadLine();
+                // useful for debugging
+                TestContext.Progress.WriteLine(outputLine);
+
+                if (ProxyPortHttp == null && TryParsePort(outputLine, "http", out _proxyPortHttp))
+                {
+                    continue;
+                }
+
+                if (_proxyPortHttps == null && TryParsePort(outputLine, "https", out _proxyPortHttps))
+                {
+                    continue;
+                }
+            }
+
+            if (_proxyPortHttp == null || _proxyPortHttps == null)
+            {
+                throw new TestProxyException("Failed to start the test proxy.");
+            }
+
+            // we need to use https when talking to test proxy admin endpoint so that we can establish the connection before any
+            // test related traffic happens which would send a Connection header for the first request. This can be switched to HTTP
+            // once https://github.com/Azure/azure-sdk-tools/issues/2303 is fixed
+            var handler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (_, certificate, _, _) => certificate.Issuer == TestProxy.DevCertIssuer
+            };
+            var options = new TestProxyClientOptions
+            {
+                Transport = new HttpClientTransport(handler)
+            };
+            Client = new TestProxyRestClient(
+                new ClientDiagnostics(options),
+                HttpPipelineBuilder.Build(options),
+                new Uri($"https://{IpAddress}:{_proxyPortHttps}"));
```

Comment:
Use HTTP and avoid cert issues?

Reply:
See the comment above - https://github.com/Azure/azure-sdk-for-net/pull/25181/files/616287c8b676a98fb47ee81eaa388c2fed2ecdf8#diff-2a9d0932d81cfbdd143b6a008f573b931c2890a5e6fece4600a9be5d11d54c24R115